gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -260,15 +757,260 @@ gst_tensor_src_iio_finalize (GObject * object)
   //FIXME: fill this function
 }
 
+/**
+ * @brief write the string in to the file
+ * @param[in] self Tensor src IIO object
+ * @param[in] file Destination file for the data
+ * @param[in] base_dir Directory containing the file
+ * @param[in] contents Data to be written to the file
+ * @return True if write was successful, false on failure
+ */
+static gboolean
+gst_tensor_write_sysfs_string (GstTensorSrcIIO * self, const gchar * file,
+    const gchar * base_dir, const gchar * contents)
+{
+  gchar *filename = NULL;
+  GError *error = NULL;
+  gboolean ret = FALSE;
+
+  filename = g_build_filename (base_dir, file, NULL);
+  if (!g_file_set_contents (filename, contents, strlen (contents), &error)) {
```

Comment:
Can g_file_set_contents() create problems here?
g_file_set_contents() recreates the file, removing existing file related metadata.

Reply:
using fprintf() instead.